src/components/config_profile/src/profile.cc

```diff
@@ -1253,6 +1223,74 @@ const std::string Profile::hmi_origin_id() const {
 void Profile::UpdateValues() {
   SDL_LOG_AUTO_TRACE();
 
+  FILE* config_file_ = fopen(config_file_name_.c_str(), "r");
+  config_obj_ = smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+  if (nullptr != config_file_) {
+    char line[INI_LINE_LEN] = "\0";
+    std::string chapter;
+
+    while (nullptr != fgets(line, INI_LINE_LEN, config_file_)) {
+      std::string line_str(line);
+      auto whitespace_len = line_str.find_first_not_of(" \t\r\n");
+      if (std::string::npos == whitespace_len) {
+        continue;
+      }
+      line_str.erase(0, whitespace_len);
+
+      if (';' == line_str[0] || '*' == line_str[0]) {
+        continue;
+      }
+
+      if ('[' == line_str[0] && std::string::npos != line_str.find(']')) {
+        line_str.erase(0, line_str.find_first_not_of("[ \t\r\n"));
+        line_str.erase(line_str.find_last_not_of(" \t\r\n]") + 1);
+
+        if (false == line_str.empty()) {
+          chapter = line_str;
+          config_obj_[chapter] =
+              smart_objects::SmartObject(smart_objects::SmartType_Map);
+        }
+
+        continue;
+      }
+
+      size_t equals_idx = line_str.find('=');
+      if (std::string::npos == equals_idx) {
+        continue;
+      }
+
+      std::string key(line_str.data(), equals_idx);
+      std::string value(line_str.data() + equals_idx + 1);
+
+      auto key_whitespace_len = key.find_first_not_of(" \t\r\n");
+      if (std::string::npos == key_whitespace_len) {
+        continue;
+      }
+      key.erase(0, key_whitespace_len);
+      key.erase(key.find_last_not_of(" \t\r\n") + 1);
+
+      if (true == config_obj_[chapter].keyExists(key)) {
+        continue;
+      }
+
+      auto val_whitespace_len = value.find_first_not_of(" \t\r\n");
+      if (std::string::npos == val_whitespace_len) {
+        value = "";
+      } else {
+        value.erase(0, value.find_first_not_of(" \t\r\n"));
+        value.erase(value.find_last_not_of(" \t\r\n") + 1);
+      }
+
+      config_obj_[chapter][key] = value;
+    }
+  }
+
+  if (nullptr != config_file_) {
+    fclose(config_file_);
+    config_file_ = nullptr;
+  }
```

Comment:
This should probably still be it's own function

Reply:
Open/Close functions were restored in https://github.com/smartdevicelink/sdl_core/pull/3688/commits/a03c69403fe14790e6b6738a65820c1ae69e95c5

Reply:
Ah, I was referring to the file parsing code

Reply:
my bad, https://github.com/smartdevicelink/sdl_core/pull/3688/commits/879fb05d55995b2aeb6054241e509161dedb85c4 was reverted and https://github.com/smartdevicelink/sdl_core/pull/3688/commits/aca5d37da826f7b610bc400060304ad2831fc7f1 moves the ini parsing into its own function